pkg/controllers/management/usercontrollers/controller.go

```diff
@@ -63,10 +70,239 @@ func (c *ClusterLifecycleCleanup) Create(obj *v3.Cluster) (*v3.Cluster, error) {
+				time.Sleep(time.Duration(sleepTime) * time.Millisecond)
+				sleepTime *= 2
+				continue
+			}
+			return err
+		}
+		break
+	}
+
+	err = userContext.Core.Namespaces("").Delete("cattle-system", &metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *ClusterLifecycleCleanup) createCleanupClusterRole(userContext *config.UserContext) (*rbacV1.ClusterRole, error) {
+	meta := metav1.ObjectMeta{
+		GenerateName: "cattle-cleanup-",
+	}
+
+	rules := []rbacV1.PolicyRule{
+		// This is needed to check for cattle-system, remove finalizers and delete
+		rbacV1.PolicyRule{
+			Verbs:     []string{"list", "get", "update", "delete"},
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+		},
+		// This might not be needed
+		rbacV1.PolicyRule{
+			Verbs:     []string{"list", "get", "delete"},
+			APIGroups: []string{""},
+			Resources: []string{"serviceaccounts"},
+		},
+		rbacV1.PolicyRule{
+			Verbs:     []string{"list", "get", "delete"},
+			APIGroups: []string{"rbac.authorization.k8s.io"},
+			Resources: []string{"roles", "rolebindings", "clusterroles", "clusterrolebindings"},
+		},
+		// The job is going to delete itself after running to trigger ownerReference
+		// cleanup of the clusterRole, serviceAccount and clusterRoleBinding
+		rbacV1.PolicyRule{
+			Verbs:     []string{"list", "get", "delete"},
+			APIGroups: []string{"batch"},
+			Resources: []string{"jobs"},
+		},
+	}
+	clusterRole := rbacV1.ClusterRole{
+		ObjectMeta: meta,
+		Rules:      rules,
+	}
+	return userContext.K8sClient.RbacV1().ClusterRoles().Create(&clusterRole)
+}
+
+func (c *ClusterLifecycleCleanup) createCleanupServiceAccount(userContext *config.UserContext) (*coreV1.ServiceAccount, error) {
+	meta := metav1.ObjectMeta{
+		GenerateName: "cattle-cleanup-",
+		Namespace:    "default",
+	}
+	serviceAccount := coreV1.ServiceAccount{
+		ObjectMeta: meta,
+	}
+	return userContext.K8sClient.CoreV1().ServiceAccounts("default").Create(&serviceAccount)
+}
+
+func (c *ClusterLifecycleCleanup) createCleanupClusterRoleBinding(userContext *config.UserContext, role, sa string) (*rbacV1.ClusterRoleBinding, error) {
+	meta := metav1.ObjectMeta{
+		GenerateName: "cattle-cleanup-",
+		Namespace:    "default",
+	}
+	clusterRoleBinding := rbacV1.ClusterRoleBinding{
+		ObjectMeta: meta,
+		Subjects: []rbacV1.Subject{
+			rbacV1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      sa,
+				Namespace: "default",
+			},
+		},
+		RoleRef: rbacV1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     role,
+		},
+	}
+	return userContext.K8sClient.RbacV1().ClusterRoleBindings().Create(&clusterRoleBinding)
+}
+
+func (c *ClusterLifecycleCleanup) createCleanupJob(userContext *config.UserContext, sa string) (*batchV1.Job, error) {
+	meta := metav1.ObjectMeta{
+		GenerateName: "cattle-cleanup-",
+		Namespace:    "default",
+		Labels:       map[string]string{"ranchercreated": "true"},
```

Comment:
do you think it would make sense to create a const for the label name?

Reply:
Yes, will add to norman